indico/modules/attachments/controllers/event_package.py

```diff
@@ -110,13 +125,23 @@ def _filter_by_sessions(self, session_ids, added_since):
         query = self._build_base_query(added_since).filter(AttachmentFolder.link_type.in_([LinkType.session,
                                                                                            LinkType.contribution,
                                                                                            LinkType.subcontribution]))
-        return [attachment for attachment in query if attachment.folder.object.session.id in session_ids]
+        return [attachment for attachment in query if attachment.folder.object.id in session_ids]
 
     def _filter_by_contributions(self, contribution_ids, added_since):
-        query = self._build_base_query(added_since).filter(AttachmentFolder.contribution_id.in_(contribution_ids),
-                                                           AttachmentFolder.link_type.in_([LinkType.contribution,
+        query = self._build_base_query(added_since).filter(AttachmentFolder.link_type.in_([LinkType.contribution,
                                                                                            LinkType.subcontribution]))
-        return [attachment for attachment in query if _get_start_dt(attachment.folder.object) is not None]
+        objs = []
+        for attachment in query:
+            linked_obj = attachment.folder.object
+            if linked_obj.is_deleted:
```

Comment:
I think this won't catch the case where the contribution is deleted but the attachment is for a subcontribution.